client/signup/jetpack-connect/index.jsx

```diff
@@ -199,6 +199,10 @@ class JetpackConnectMain extends Component {
 			return false;
 		}
 
+		if ( this.checkProperty( 'isJetpackConnected' ) ) {
```

Comment:
Thinking out loud and this is outside the scope of this PR, but it would be nice to rethink this whole section.